### ISSUE TITLE
moveit2 repo is missing in moveit2.repo

### ISSUE THREAD

**Issue by ipa-hsd** [ros-planning/moveit2#403]
### Description

[moveit2.repos](https://github.com/ros-planning/moveit2/blob/main/moveit2.repos) is missing an entry for `moveit2` repo. [Dockerfile](https://github.com/ros-planning/moveit2/blob/main/.docker/source/Dockerfile#L17) expects that `moveit2` repo is imported as well. The [installation guide](https://moveit.ros.org/install-moveit2/source/) (`Download Source Code` section) mentions a separate step to clone the repo.

### Your environment
* ROS Distro: Foxy
* OS Version: e.g. Ubuntu 20.04
* Source - main

### Steps to reproduce
```
docker pull moveit/moveit2:foxy-source
docker run -it moveit/moveit2:foxy-source
```

### Expected behaviour
This should have the `moveit2` packages in the `install` folder.

### Actual behaviour
`moveit2` packages are missing in `moveit/moveit2:foxy-source` image.

**Comment by tylerjw** [comment]
I should have fixed this when I changed what is in the repos file.  We are now using it in the CI system to declare the upstream dependencies so we need to get moveit2 into the docker image some other way.

**Comment by tylerjw** [comment]
You should be able to test (my PR) this locally like this:

```
docker build https://github.com/tylerjw/moveit2.git\#fix-source-dockerfile:.docker/source -t moveit2:source
```

Then after it finishes building you can use it like this:

```
docker run -it moviet2:source /bin/bash
```

**Comment by tylerjw** [comment]
I merged it and deleted the branch.  If you still have yet to build it locally, here is the command that now points at the repo now that it is in main:

```
docker build https://github.com/ros-planning/moveit2.git\#main:.docker/source -t moveit2:source
```

**Comment by ipa-hsd** [comment]
@tylerjw thanks for the quick fix! After building the image, tested with 
```
$ ./gui-docker -it --rm --network host moveit2:source /bin/bash
# ros2 launch run_moveit_cpp run_moveit_cpp.launch.py
```
and the demo works as expected! 

But why can't the `moveit2` repo be added to `moveit2.repos`? `vcs` already does the job of cloning, so the additional step of explicitly cloning `moveit2` won't be necessary.

**Comment by tylerjw** [comment]
> But why can't the `moveit2` repo be added to `moveit2.repos`? `vcs` already does the job of cloning, so the additional step of explicitly cloning `moveit2` won't be necessary.

Before last week we had custom logic to handle using the moveit2.repos file with moveit2 in it in the CI system.  When I migrated us to GitHub Actions I also migrated us away from our custom code.  The issue is that the repos file specifies a specific branch to checkout for moveit and when you run CI in a PR you want the branch that has the changes from the PR in it.  Secondly the repos file is for specifying source dependencies.  If you include moveit2 in there you create circular source dependencies.
